Patch rubygem-safemode to allow to_sentence for array

dgoetz opened the pull request:
Safemode has not been released in a while but now includes a fix I need, so I opted for bumping to snapshot version.

ekohl commented:
Have you asked for a release? IIRC we also have various people with commit access and https://rubygems.org/gems/safemode suggests 2 former devs also have release access.

dgoetz commented:
I did not ask explicitly, only implied in my pull request which fixed my problem. Shall I open an issue to have a new release over there?

ekohl commented:
Please do. If it isn't addressed, we can link to it.

dgoetz commented:
Ok, lets see if we get a new release: https://github.com/svenfuchs/safemode/issues/33

dgoetz commented:
No release yet for the safemode gem. 
@ekohl How do you want to progress? Should I add a note to the spec or is the release version good enough as indicator?

ekohl commented:
Given the lack of response upstream, I'm ok with proceeding here.

On the technical details: where does it get the tarball from? Is the preview tarball on Rubygems?

dgoetz commented:
The gem is not on rubygems. It is a `gem build` on the date of the commit that is indicated by the snapshot version in the package release, so `20200714gitd9b1630` stands for commit `d9b1630` and build date `14.07.2020`. 

I have no idea if there is a better way for ruby gems, because for most other software I would get the snapshot tarball without a command to run locally.

ekohl commented:
So normally we get the actual gem via git annex. It uses `Source0` which points to Rubygems. It also doesn't refer to the pre version so I'm wondering if it actually has the patched version. If it doesn't, that may indicate a problem in our build chain.

dgoetz commented:
I checked the gem in via git annex, so if it uses this it would be the patched version. I locally build the package and have it already installed on a productive environment where the patch originates from. 

The packages are removed from koji output, so I can not verify koji build packages at the moment. Shall I rerun the tests?

Another approach would be using the release gem and then apply patches. Not sure if this actually works, but I could give it a try. Would this be preferred?

ekohl commented:
[test rpm] since the scratch builds have been rotated.

dgoetz commented:
Ok, it seems like it is not taking the source from git annex (package is missing the to_sentence from the fix). So I will try to use the original gem and a patch.

dgoetz commented:
@ekohl Not sure about versioning, commit message and if patch should go to git annex. If I am correct, strictly by the rule is adding a patch only a new package release, or should it be named differently? Are commit message and changelog entry ok? Should the patch also go to git annex like the gems?

ekohl commented:
My understanding is that git annex is a way of indirection, storing only a checksum of the file. We can derive it from the Source, so know it comes from Rubygems. We do this to prevent storing big binaries in git. I'm not sure how the previous build passed since my understanding is that it shouldn't have.

The patch can live on some HTTP source, but this is small enough to just inline.

For the commit message, you can consider describing the patch (`Patch safemode to allow to_sentence`). You may also want to do the same in the filename, but since we (luckily) have next to no patches, it's not that important.

dgoetz commented:
Ok, I added a comment to the spec, but left the patch name unchanged as it is available by this name via github url. Furthermore I adjusted the commit message and title of the PR. Tests are green, so everything is ready for a hopefully final review. Thanks!

evgeni commented on packages/foreman/rubygem-safemode/rubygem-safemode.spec:
```diff
@@ -7,11 +7,13 @@ Summary: A library for safe evaluation of Ruby code
 URL: https://github.com/svenfuchs/safemode
 Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+# Patch safemode to allow to_sentence for array
+Patch0: https://patch-diff.githubusercontent.com/raw/svenfuchs/safemode/pull/32.patch
```
You're not annexing the patch, so this can be just `32.patch` or even better `allow-to_sentence.patch` ;)

evgeni replied:
not blocking, as it works as it is

dgoetz replied:
I wanted to keep the URL so it is easily recognizable that the patch is upstream.
